python_modules/dagster/dagster/_core/definitions/repository_definition/repository_data_builder.py

```diff
@@ -247,37 +279,38 @@ def build_caching_repository_data_from_list(
     # resolve all the UnresolvedAssetJobDefinitions using the full set of assets
     if unresolved_jobs:
         for name, unresolved_job_def in unresolved_jobs.items():
-            resolved_job = unresolved_job_def.resolve(
-                asset_graph=asset_graph,
-                default_executor_def=default_executor_def,
-                resource_defs=top_level_resources,
+            jobs[name] = lambda: _resolve_unresolved_job_def(
+                unresolved_job_def,
+                asset_graph,
+                default_executor_def,
+                top_level_resources,
+                default_logger_defs,
             )
-            jobs[name] = resolved_job
 
     # resolve all the UnresolvedPartitionedAssetScheduleDefinitions using
     # the resolved job containing the partitions def
+
+    unresolved_schedules = {}
+
     if unresolved_partitioned_asset_schedules:
         for (
             name,
             unresolved_partitioned_asset_schedule,
         ) in unresolved_partitioned_asset_schedules.items():
-            resolved_job = jobs[unresolved_partitioned_asset_schedule.job.name]
-            schedules[name] = unresolved_partitioned_asset_schedule.resolve(
-                cast(JobDefinition, resolved_job)
+            job = jobs[unresolved_partitioned_asset_schedule.job.name]
+            unresolved_schedules[name] = lambda _: unresolved_partitioned_asset_schedule.resolve(
+                cast(JobDefinition, job() if isfunction(job) else job)
             )
```

Comment:
this is unfortunate in that it results in the job getting potentially resolved twice - once when loading the job and once when loading the schedule. would need to be able to pull from the CachingRepositoryData cache while resolve this somehow in order to resolve that

Reply:
Or maybe doing this business within CachingRepositoryData rather than here

Reply:
Moved to CachingRepositoryData